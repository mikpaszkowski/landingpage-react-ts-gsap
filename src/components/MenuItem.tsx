import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../styles/styleConstants";

const MenuItemWrapper = styled.div`
  margin: 0.7rem 0;
`;

const MenuItemText = styled.p`
  font-family: ${fontTypes.secondaryFont};
  font-style: normal;
  font-weight: normal;
  font-size: 2.4rem;
  line-height: 3.6rem;
`;

interface MenuItemProps {
  name: string;
}

export const MenuItem = ({ name }: MenuItemProps) => {
  return (
    <MenuItemWrapper>
      <MenuItemText>{name}</MenuItemText>
    </MenuItemWrapper>
  );
};
