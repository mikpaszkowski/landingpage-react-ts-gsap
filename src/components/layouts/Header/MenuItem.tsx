import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../../styles/styleConstants";
import { ReactComponent as FocusIcon } from "./ellipse-4.svg";

const MenuItemWrapper = styled.div`
  margin: 0.7rem 0;
`;

const MenuItemText = styled.a`
  position: relative;
  font-family: ${fontTypes.secondaryFont};
  font-style: normal;
  font-weight: normal;
  font-size: 2.4rem;
  line-height: 3.6rem;
  text-decoration: none;
  color: ${({ theme }) => theme.palette.primary.font};

  &:focus > svg {
    display: block;
  }
`;

const FocusSVG = styled(FocusIcon)`
  display: none;
  position: absolute;
  bottom: -0.8rem;
  left: 50%;
  transform: translateX(-50%);
`;

interface MenuItemProps {
  name: string;
}

export const MenuItem = ({ name }: MenuItemProps) => {
  return (
    <MenuItemWrapper>
      <MenuItemText href="#">
        {name}
        <FocusSVG />
      </MenuItemText>
    </MenuItemWrapper>
  );
};
