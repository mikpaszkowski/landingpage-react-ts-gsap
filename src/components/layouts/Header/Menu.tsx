import * as React from "react";
import styled from "styled-components";
import { MenuItem } from "./MenuItem";
import { CustomButton } from "../../ReusableComponents/CustomButton";
import { IMenuItems } from "../../../assets/menuItems/menuItems";
import { device } from "../../../styles/responsive";

const MenuWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 64.4rem;

  @media ${device.mobileMedium} {
    display: none;
  }
`;
interface IMenuProps {
  menuItems: Array<IMenuItems>;
}

export const Menu = ({ menuItems }: IMenuProps) => {
  return (
    <MenuWrapper>
      {menuItems.map(({ id, name }) => (
        <MenuItem key={id} name={name} />
      ))}

      <CustomButton fontSize="2.4rem" lineHeight="3.6rem" name="Log in" />
    </MenuWrapper>
  );
};
