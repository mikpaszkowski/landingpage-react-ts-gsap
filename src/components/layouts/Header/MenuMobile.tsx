import * as React from "react";
import styled, { keyframes } from "styled-components";
import { MenuItem } from "./MenuItem";
import { CustomButton } from "../../ReusableComponents/CustomButton";
import { menuItems, IMenuItems } from "../../../assets/menuItems/menuItems";
import { IoCloseOutline } from "react-icons/io5";

const slideDown = keyframes`
  from{
    transform: translateY(-100%);
  }
  to{
    transform: translateY(0);
  }
`;

const MenuWrapper = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  z-index: 999;
  padding: 7rem 2.4rem 5rem 2.4rem;
  background-color: ${({ theme }) => theme.palette.common.background};
  transform: translateY(0);
  animation-name: ${slideDown};
  animation-duration: 0.4s;
  animation-timing-function: cubic-bezier(0.445, 0.05, 0.55, 0.95);
  animation-fill-mode: both;
  transition: all 1s ease-in-out;
`;

const Menu = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  height: 100%;
`;

const LogInButton = styled(CustomButton)`
  margin-top: 4rem;
  width: 100%;
`;

const CloseButton = styled(IoCloseOutline)`
  position: absolute;
  top: 2.7rem;
  right: 2.4rem;
  width: 5rem;
  height: 5rem;
  cursor: pointer;
  transition: transform 0.3s ease-in-out;

  &:hover {
    transform: rotate(90deg);
  }
`;

interface MenuMobileProps {
  isMenuOpen: boolean | null;
  closeMenuMobile: () => void;
  menuItems: Array<IMenuItems>;
}

export const MenuMobile = ({
  isMenuOpen,
  closeMenuMobile,
  menuItems,
}: MenuMobileProps) => {
  return isMenuOpen ? (
    <MenuWrapper>
      <Menu>
        {menuItems.forEach(({ id, name }) => {
          <MenuItem key={id} name={name} />;
        })}
        <LogInButton fontSize="2.4rem" lineHeight="3.6rem">
          Log in
        </LogInButton>
      </Menu>
      <CloseButton onClick={closeMenuMobile} />
    </MenuWrapper>
  ) : null;
};
