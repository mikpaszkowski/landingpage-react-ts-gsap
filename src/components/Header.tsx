import * as React from "react";
import { Logo } from "./Logo";
import { Menu } from "./Menu";
import styled from "styled-components";
import { ReactComponent as Bar } from "../assets/images/svg-images/menu-bar.svg";
import { device } from "../styles/responsive";
import { MenuMobile } from "./MenuMobile";
import { colors } from "../styles/styleConstants";

const HeaderWrapper = styled.header`
  position: fixed;
  top: 0;
  left: 50%;
  z-index: 10;
  max-width: 145rem;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin: 0 auto;
  padding: 2.7rem 16.7rem;
  transform: translateX(-50%);
  background-color: ${colors.background};

  @media ${device.mobileMedium} {
    position: relative;
    padding: 2.7rem 2.4rem;
  }
`;
const MenuBar = styled(Bar)`
  display: none;
  width: 3rem;
  height: 1.7rem;
  cursor: pointer;

  @media ${device.mobileMedium} {
    display: block;
    position: absolute;
    top: 50%;
    right: 2.4rem;
    transform: translateY(-50%);
  }
`;

const { useState } = React;

export const Header = () => {
  const [menuBarOpen, setMenuBarOpen] = useState<boolean | null>(null);

  const handleClick = () => {
    setMenuBarOpen(!menuBarOpen);
  };

  return (
    <HeaderWrapper>
      <Logo />
      <Menu />
      <MenuBar onClick={handleClick} />
      <MenuMobile isMenuOpen={menuBarOpen} closeMenuMobile={handleClick} />
    </HeaderWrapper>
  );
};
