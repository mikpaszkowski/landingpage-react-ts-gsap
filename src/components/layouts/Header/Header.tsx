import * as React from "react";
import { Logo } from "../../ReusableComponents/Logo/Logo";
import { Menu } from "./Menu";
import styled from "styled-components";
import { ReactComponent as Bar } from "./menu-bar.svg";
import { menuItems } from "../../staticContents/menu/menuItems";
import { device } from "../../../styles/responsive";
import { MenuMobile } from "./MenuMobile";

const HeaderWrapper = styled.header`
  position: fixed;
  top: 0;
  left: 50%;
  z-index: 9999;
  max-width: 145rem;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin: 0 auto;
  padding: 2.7rem 16.7rem;
  transform: translateX(-50%);
  background-color: ${({ theme }) => theme.palette.common.background};

  @media ${device.laptopSmall} {
    padding: 2.7rem 6rem;
  }

  @media ${device.mobileMedium} {
    position: relative;
    padding: 2.7rem 2.4rem;
  }
`;
const MenuBar = styled(Bar)`
  display: none;
  width: 3rem;
  height: 1.7rem;
  cursor: pointer;
  fill: ${({ theme }) => theme.palette.primary.font};

  @media ${device.tabletLarge} {
    display: block;
    position: absolute;
    top: 50%;
    right: 2.4rem;
    transform: translateY(-50%);
    fill: ${({ theme }) => theme.palette.primary.font};
  }
`;

const { useState } = React;

export const Header = () => {
  const [menuBarOpen, setMenuBarOpen] = useState<boolean | null>(null);

  const handleClick = () => {
    setMenuBarOpen(!menuBarOpen);
  };

  return (
    <HeaderWrapper>
      <Logo />
      <Menu menuItems={menuItems} />
      <MenuBar onClick={handleClick} />
      <MenuMobile
        isMenuOpen={menuBarOpen}
        closeMenuMobile={handleClick}
        menuItems={menuItems}
      />
    </HeaderWrapper>
  );
};
