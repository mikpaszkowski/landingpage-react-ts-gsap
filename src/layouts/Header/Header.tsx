import * as React from "react";
import { Logo } from "../../components/Logo/Logo";
import { Menu } from "./Menu";
import styled from "styled-components";
import { useTheme } from "../../hooks/useTheme";
import { lightTheme } from "../../styles/theme";
import { ReactComponent as BlackBar } from "../../assets/svg/menu-bar-black.svg";
import { ReactComponent as WhiteBar } from "../../assets/svg/menu-bar-black.svg";
import { menuItems } from "../../data/staticContents/menu/menuItems";
import { device } from "../../styles/responsive";
import { MenuMobile } from "./MenuMobile";

const HeaderWrapper = styled.header`
  position: fixed;
  top: 0;
  left: 50%;
  z-index: 9999;
  max-width: 145rem;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin: 0 auto;
  padding: 2.7rem 16.7rem;
  transform: translateX(-50%);
  background-color: ${({ theme }) => theme.palette.common.background};

  @media ${device.laptopSmall} {
    padding: 2.7rem 6rem;
  }

  @media ${device.mobileMedium} {
    position: relative;
    padding: 2.7rem 2.4rem;
  }
`;
const BlackMenuBar = styled(BlackBar)`
  display: none;
  width: 3rem;
  height: 1.7rem;
  cursor: pointer;

  @media ${device.tabletLarge} {
    display: block;
    position: absolute;
    top: 50%;
    right: 2.4rem;
    transform: translateY(-50%);
  }
`;

const WhiteMenuBar = styled(WhiteBar)`
  display: none;
  width: 3rem;
  height: 1.7rem;
  cursor: pointer;
  color: white;

  svg {
    fill: white;
  }

  @media ${device.tabletLarge} {
    display: block;
    position: absolute;
    top: 50%;
    right: 2.4rem;
    transform: translateY(-50%);
    fill: ${({ theme }) => theme.palette.primary.font};
  }
`;

const { useState } = React;

export const Header = () => {
  const [menuBarOpen, setMenuBarOpen] = useState<boolean | null>(null);
  const { theme } = useTheme();

  const handleClick = () => {
    setMenuBarOpen(!menuBarOpen);
  };

  return (
    <HeaderWrapper>
      <Logo />
      <Menu menuItems={menuItems} />
      {theme === lightTheme ? (
        <BlackMenuBar onClick={handleClick} />
      ) : (
        <WhiteMenuBar onClick={handleClick} />
      )}
      <MenuMobile
        isMenuOpen={menuBarOpen}
        closeMenuMobile={handleClick}
        menuItems={menuItems}
      />
    </HeaderWrapper>
  );
};
