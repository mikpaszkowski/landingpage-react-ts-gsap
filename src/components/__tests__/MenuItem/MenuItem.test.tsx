import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import { renderWithLightTheme, mountWithDarkTheme } from "../helpers/index";
import { MenuItem } from "../../../layouts/Header/MenuItem";

afterEach(() => {
  cleanup();
});

test("check render correctly <MenuItem /> component", () => {
  const tree = renderWithLightTheme(<MenuItem name="about" />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("check MenuItem component", () => {
  const wrapper = mountWithDarkTheme(<MenuItem name="Contact" />);

  test("should contain prop name", () => {
    expect(wrapper.find(MenuItem).prop("name")).toEqual("Contact");
  });

  test("should contain a tag", () => {
    expect(wrapper.find("a").isEmptyRender()).toEqual(false);
  });
});
