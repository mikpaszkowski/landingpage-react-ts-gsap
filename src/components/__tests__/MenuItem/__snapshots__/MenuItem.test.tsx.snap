// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render correctly <MenuItem /> component 1`] = `
.c0 {
  margin: 0.7rem 0;
}

.c1 {
  position: relative;
  font-family: 'Reem Kufi',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 2.4rem;
  line-height: 3.6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #131E07;
}

.c1:focus > svg {
  display: block;
}

.c2 {
  display: none;
  position: absolute;
  bottom: -0.8rem;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

<div
  className="c0"
>
  <a
    className="c1"
    href="#"
  >
    about
    <svg
      className="c2"
    >
      ellipse-4.svg
    </svg>
  </a>
</div>
`;
